statsd/options.go

```diff
@@ -26,6 +26,10 @@ var (
 	DefaultWriteTimeoutUDS = 1 * time.Millisecond
 	// DefaultTelemetry is the default value for the Telemetry option
 	DefaultTelemetry = true
+	// DefaultSendingMode is the default behavior when sending metrics
+	DefaultSendingMode = BlockOnSend
+	// DefaultDropSendingModeBufferSizer is the default size of the channel holding incoming metrics
+	DefaultDropSendingModeBufferSizer = 2048
```

Comment:
Did you test the drop rate with a rate limited client ? We usually see about 16 metrics per UDP payload. Since the client currently holds 2048 buffers of ~16 metrics by default we will start dropping 16 times earlier

Reply:
Not sure I understand the comment. The `DefaultDropSendingModeBufferSizer` is for the input buffer before the worker (so before batching). That buffer holds up to 2048 single metrics at a time by default.

A rate limited client would drop less metrics since it would leave time to the workers to consume the metrics.

Reply:
What I meant was that 2048 single metrics is much lower than what we have today in the next pipe. I am a bit worried it's going to fill up fast even at relatively low MPS. I mentioned a rate limiter because the current benchmarks only tells us the drop % when you fully saturate the client. I think it would be more interesting to get data at a few fixed rates.

Reply:
I used this because that's what  we have internally in our wrapper around dogatstsd (that implement the channel mode).

We could bump it to `4096` but knowing that we have a channel per worker that means 32*2048 metrics. Of course if a a client sends the same metrics the worker sharding doesn't help. I'm worried of the default memory consumption of having a higher default. WDYT ?

Reply:
Merging for now to unlock client side aggregation. We can change the default later.